### PR TITLE
[FLINK-9523] [table] Add Kafka examples for Flink Table/SQL API

### PR DESCRIPTION
**Pull request by SokolovMS**
## What is the purpose of the change

Implementation of FLINK-9523

## Brief change log

Added an example of using Kafka JSON TableSource with executing SQL queries on input table.

## Verifying this change

1. Start ZooKeeper
2. Start Kafka
3. Run Consumer with arguments specified in the JavaDoc
4. Run Producer with arguments specified in the JavaDoc

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no. Could only use them.
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

### CODE REVIEW

**Comment by SokolovMS** [comment]
Found the following problems:
1. Jenkins failed due to checkstyle plugin fail. 
_src/main/java/org/apache/flink/configuration/MetricOptions.java:[22,8] (imports) UnusedImports: Unused import: org.apache.flink.configuration.description.Description._
But I didn't touch mentioned file.
2. When I start Kafka Producer I have an exception
_org.apache.flink.util.FlinkException: The assigned slot b6257dd9-f294-4cd7-beb5-fb52f85c4b96_1 was removed._
Could anyone suggest what to do with it?

**Comment by yanghua** [comment]
hi @SokolovMS thanks for your contribution. Firstly, please fix the checkstyle error: 

```
ERROR] src/main/java/org/apache/flink/configuration/MetricOptions.java:[22,8] (imports) UnusedImports: Unused import: org.apache.flink.configuration.description.Description.
```

**Comment by SokolovMS** [comment]
Merged from master. FlinkException disappeared.

File: flink-examples/flink-examples-table/src/main/java/org/apache/flink/table/examples/java/kafka/Kafka010JsonConsumer.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+				"Usage: Kafka --input-topic <topic> --output-topic <topic> " +
+				"--bootstrap.servers <kafka brokers> " +
+				"--zookeeper.connect <zk quorum> --group.id <some id>");
+			return;
+		}
+
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		StreamTableEnvironment tEnv = StreamTableEnvironment.getTableEnvironment(env);
+
+		Kafka010JsonTableSource sensorSource = Kafka010JsonTableSource.builder()
```
Comment (twalthr) [inline flink-examples/flink-examples-table/src/main/java/org/apache/flink/table/examples/java/kafka/Kafka010JsonConsumer.java]: You are using outdated API. Please use the new APIs as described here:
https://ci.apache.org/projects/flink/flink-docs-master/dev/table/connect.html

File: flink-examples/flink-examples-table/src/main/java/org/apache/flink/table/examples/java/kafka/Kafka010JsonConsumer.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+ *
+ * <p>This example shows how to:
+ *  - Use Kafka JSON TableSource
+ *  - Use SQL api and a group window to compute a result
+ *
+ * <p>Example usage:
+ * Run Consumer with the following arguments:
+ * 	--input-topic consumer-input --output-topic consumer-output --bootstrap.servers localhost:9092 --zookeeper.connect localhost:2181 --group.id myconsumer
+ */
+public class Kafka010JsonConsumer {
```
Comment (twalthr) [inline flink-examples/flink-examples-table/src/main/java/org/apache/flink/table/examples/java/kafka/Kafka010JsonConsumer.java]: I would call the classes `Kafka010JsonExample` to differentiate from actual consumer/producer classes.

File: flink-examples/flink-examples-table/src/main/java/org/apache/flink/table/examples/java/kafka/Kafka010JsonConsumer.java (hunk 1 above)
Comment (SokolovMS) [reply flink-examples/flink-examples-table/src/main/java/org/apache/flink/table/examples/java/kafka/Kafka010JsonConsumer.java]: Should I use updated API for creating Kafka sink?

File: flink-examples/flink-examples-table/src/main/java/org/apache/flink/table/examples/java/kafka/Kafka010JsonConsumer.java (hunk 2 above)
Comment (SokolovMS) [reply flink-examples/flink-examples-table/src/main/java/org/apache/flink/table/examples/java/kafka/Kafka010JsonConsumer.java]: Done

File: flink-examples/flink-examples-table/src/main/java/org/apache/flink/table/examples/java/kafka/Kafka010JsonConsumer.java (hunk 1 above)
Comment (SokolovMS) [reply flink-examples/flink-examples-table/src/main/java/org/apache/flink/table/examples/java/kafka/Kafka010JsonConsumer.java]: Done for Kafka source

**Comment by SokolovMS** [comment]
@twalthr please continue the review.

**Comment by SokolovMS** [comment]
@twalthr Could you please take a look again on that PR?
Thank you

**Comment by aljoscha** [comment]
I'm closing this as "Abandoned", since there is no more activity and the code base has moved on quite a bit. The interfaces of the Table API and how we interface with connectors have changed a log. Please re-open a new PR if you still want to work on this Jira issue.

**Pull request closed without merging**